### PR TITLE
Related to #1004: Fixed jackson dependency.

### PR DESCRIPTION
**Pull request by MrK191**

### CODE REVIEW

File: dataverse-webapp/pom.xml [hunk 1]
```diff
@@ -146,11 +146,13 @@
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-core</artifactId>
+            <version>2.5.1</version>
```
Comment (kaczynskid) [inline dataverse-webapp/pom.xml]: We're managing `Jackson` version in base `pom.xml` using BOM in version `2.9.6`. Why do you try to override to a different version here?

**Review by kaczynskid — CHANGES_REQUESTED** [review]

File: dataverse-webapp/pom.xml (hunk 1 above)
Comment (MrK191) [reply dataverse-webapp/pom.xml]: There was no BOM version in `2.5.1` but after testing with different versions now I can see that `2.8.3` is also not causing the crash.

File: pom.xml [hunk 2]
```diff
@@ -31,7 +31,7 @@
         </argLine>
 
         <aws.version>1.11.538</aws.version>
-        <jackson.version>2.9.6</jackson.version>
+        <jackson.version>2.8.3</jackson.version>
```
Comment (kaczynskid) [inline pom.xml]: Why `2.8.3`? Why not latest `2.11.0`? Or `2.8.11.20200310`, latest from `2.8.x`? What's the reasoning be behind previous `2.5.1` or this `2.8.3` specific versions?

Comment (MrK191) [reply pom.xml]: `2.5.1` is the version on glassfish, `2.8.3` is the lowest version that compiles and works, I could try each version to find the highest one but the higher version the higher the chance that it will conflict with glassfish version.

Comment (kaczynskid) [reply pom.xml]: Ah, now I understand. In that case, `2.5.1` in `provided` scope seems the way to go. Since there is no `BOM` for that version, I'd remove it entirely and modify `<dependencyManagement>` section with specific dependencies. So mostly what you did in the first place, just in a different place.

And I'd say in the commit message sth like:

> Aligned Jackson version to match the one on Glassfish 4.2.1

as just saying `fixed` does not explain things well enough.

File: pom.xml [hunk 3]
```diff
@@ -242,6 +247,7 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <scope>test</scope>
+            <version>${junit.version}</version>
```
Comment (kaczynskid) [inline pom.xml:250]: I didn't realize we have both Junit4 & Junit5 on the classpath. We should only have Junit5, as it has backwards compatibility features for Junit4 and Junit3.

File: dataverse-webapp/pom.xml [hunk 4]
```diff
@@ -142,16 +142,6 @@
...
-        <!-- Should be refactored and removed once on Java EE 8 -->
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-core</artifactId>
-        </dependency>
-        <!-- Should be refactored and removed once on Java EE 8 -->
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-databind</artifactId>
-        </dependency>
```
Comment (kaczynskid) [inline dataverse-webapp/pom.xml]: Now I'm lost again... Why are those removed, instead of only changed into `provided` scope? We didn't actually use them?

File: pom.xml (hunk 3 above)
Comment (MrK191) [reply pom.xml:250]: The problem is that we are using a method that is not present without it like `assertNotEquals`.

File: dataverse-webapp/pom.xml (hunk 4 above)
Comment (MrK191) [reply dataverse-webapp/pom.xml]: Well since I added correct dependency in parent pom wouldn't having it again in child pom be redundant?

Comment (kaczynskid) [reply dataverse-webapp/pom.xml]: No, you only added it in `<dependencyManagement>`, and this here is `<dependencies>` section. In `<dependencyManagement>` we don't actually declare dependencies, but only their versions and possibly other properties for easier re-use in modules. We still need them in `<dependencies>` to actually declare the dependency it a module that actually needs it.

File: pom.xml (hunk 3 above)
Comment (kaczynskid) [reply pom.xml:250]: Yeah, but as I said, there are [backwards compatibility features](https://stackoverflow.com/questions/41635164/is-junit5-backward-compatible-to-junit4) that can be used instead of having both engines on the classpath. But it seems unrelated with the main issue, so can go to a separate PR.

File: dataverse-webapp/pom.xml (hunk 4 above)
Comment (MrK191) [reply dataverse-webapp/pom.xml]: My bad, gotta read the docs.

Comment (MrK191) [reply dataverse-webapp/pom.xml]: Btw do I need to add `provided` scope? Shouldn't it be inherited? `when a POM derived from this one declares a dependency described by a matching groupId and artifactId, the version and other values from this section are used for that dependency if they were not already specified.`

Comment (kaczynskid) [reply dataverse-webapp/pom.xml]: You already defined the `scope` in `<dependencyManagement>` section, so you don't have to repeat it here. I usually only define versions there, as one might want to use the dependency in different scopes in different contexts. That's why I mentioned it in the first comment. So no, you don't need to :)

**Review by kaczynskid — APPROVED** [review]
